src/test/java/com/github/arteam/dropwizard/http2/client/InstrumentedHttpClientTest.java

```diff
@@ -0,0 +1,85 @@
+package com.github.arteam.dropwizard.http2.client;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.github.arteam.dropwizard.http2.client.names.NameStrategies;
+import io.dropwizard.testing.ResourceHelpers;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import org.assertj.core.api.SoftAssertions;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.http.HttpClientTransportOverHTTP;
+import org.eclipse.jetty.http2.client.HTTP2Client;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class InstrumentedHttpClientTest {
+    @ClassRule
+    public static final DropwizardAppRule<TestConfiguration> rule = new DropwizardAppRule<>(TestApplication.class,
+            ResourceHelpers.resourceFilePath("server.yml"));
+
+    @Test
+    public void testInstrumentation() throws Exception {
+        final MetricRegistry metrics = new MetricRegistry();
```

Comment:
We could make `metrics` a field in `InstrumentedHttpClientTest` and reuse it across the tests.